test/unit/utils_test.js

```diff
@@ -68,6 +68,1210 @@ describe('Utils', function() {
+        .toEnvelope()
+        .toXDR("base64")
+        .toString();
+
+      expect(
+        StellarSdk.Utils.verifyChallengeTxSigners(
+          signedChallenge,
+          this.serverKP.publicKey(),
+          StellarSdk.Networks.TESTNET,
+          [this.clientKP1.publicKey()],
+        ),
+      ).to.eql([this.clientKP1.publicKey()]);
+    });
+
+    it("throws an error if the server hasn't signed the transaction", function() {
+      const transaction = new StellarSdk.TransactionBuilder(
+        this.txAccount,
+        this.txBuilderOpts,
+      )
+        .addOperation(this.operation)
+        .setTimeout(30)
+        .build();
+
+      transaction.sign(StellarSdk.Keypair.random()); // Signing with another key pair instead of the server's
+
+      const invalidsServerSignedChallenge = transaction
+        .toEnvelope()
+        .toXDR("base64")
+        .toString();
+
+      expect(() =>
+        StellarSdk.Utils.verifyChallengeTxSigners(
+          invalidsServerSignedChallenge,
+          this.serverKP.publicKey(),
+          StellarSdk.Networks.TESTNET,
+          [this.clientKP1.publicKey()],
+        ),
+      ).to.throw(
+        StellarSdk.InvalidSep10ChallengeError,
+        "Transaction not signed by server: '" + this.serverKP.publicKey() + "'",
+      );
+    });
+
+    it("throws an error if there are no clients signing the Tx", function() {
+      const challenge = StellarSdk.Utils.buildChallengeTx(
+        this.serverKP,
+        this.clientKP1.publicKey(),
+        "SDF",
+        300,
+        StellarSdk.Networks.TESTNET,
+      );
+
+      clock.tick(200);
+
+      expect(() =>
+        StellarSdk.Utils.verifyChallengeTxSigners(
+          challenge,
+          this.serverKP.publicKey(),
+          StellarSdk.Networks.TESTNET,
+          [],
+        ),
+      ).to.throw(
+        StellarSdk.InvalidSep10ChallengeError,
+        /No verifiable client signers provided, at least one G... address must be provided/,
+      );
+    });
+
+    it("throws an error if the challenge was signed by an unrecognized client", function() {
+      const challenge = StellarSdk.Utils.buildChallengeTx(
+        this.serverKP,
+        this.clientKP1.publicKey(),
+        "SDF",
+        300,
+        StellarSdk.Networks.TESTNET,
+      );
+
+      clock.tick(200);
+
+      const transaction = new StellarSdk.Transaction(
+        challenge,
+        StellarSdk.Networks.TESTNET,
+      );
+      const unrecognizedKP = StellarSdk.Keypair.random();
+      transaction.sign(unrecognizedKP);
+
+      expect(() =>
+        StellarSdk.Utils.verifyChallengeTxSigners(
+          challenge,
+          this.serverKP.publicKey(),
+          StellarSdk.Networks.TESTNET,
+          [this.clientKP1.publicKey()],
+        ),
+      ).to.throw(
+        StellarSdk.InvalidSep10ChallengeError,
+        "Transaction not signed by client(s): '" +
+          this.clientKP1.publicKey() +
+          "'",
+      );
+    });
```

Comment:
I think this test's description and its contents are a bit of a mismatch too. The expectation centers on the transaction not being signed by the known client, and now how a transaction is handled if it has unrecognized signatures. If the transaction had been signed by the client and the unrecognized signature existed, the expectation would be an error about the signature being unrecognized.

Reply:
@marcelosalloum: remember to review the error `throws an error if the challenge was signed by an unrecognized client`